casanovo/denovo/model_runner.py

```diff
@@ -0,0 +1,298 @@
+    )
+    train_loader = DeNovoDataModule(
+        train_index=train_index, **dataloader_params
+    )
+    train_loader.setup()
+    val_loader = DeNovoDataModule(valid_index=val_index, **dataloader_params)
+    val_loader.setup()
+    # Initialize the model.
+    model_params = dict(
+        dim_model=config["dim_model"],
+        n_head=config["n_head"],
+        dim_feedforward=config["dim_feedforward"],
+        n_layers=config["n_layers"],
+        dropout=config["dropout"],
+        dim_intensity=config["dim_intensity"],
+        custom_encoder=config["custom_encoder"],
+        max_length=config["max_length"],
+        residues=config["residues"],
+        max_charge=config["max_charge"],
+        n_log=config["n_log"],
+        tb_summarywriter=config["tb_summarywriter"],
+        warmup_iters=config["warmup_iters"],
+        max_iters=config["max_iters"],
+        lr=config["learning_rate"],
+        weight_decay=config["weight_decay"],
+    )
+    if config["train_from_scratch"]:
+        model = Spec2Pep(**model_params)
+    else:
+        if not os.path.isfile(model_filename):
+            logger.error(
+                "Could not find the model weights at file %s to continue "
+                "training",
+                model_filename,
+            )
+            raise FileNotFoundError(
+                "Could not find the model weights to continue training"
+            )
+        model = Spec2Pep().load_from_checkpoint(model_filename, **model_params)
+    # Create the Trainer object and (optionally) a checkpoint callback to
+    # periodically save the model.
+    if config["save_model"]:
+        callbacks = [
+            pl.callbacks.ModelCheckpoint(
+                dirpath=config["model_save_folder_path"],
+                save_top_k=-1,
+                save_weights_only=config["save_weights_only"],
+                every_n_train_steps=config["every_n_train_steps"],
+            )
+        ]
+    else:
+        callbacks = None
+    trainer = pl.Trainer(
+        accelerator="auto",
+        auto_select_gpus=True,
+        callbacks=callbacks,
+        devices=-1,
+        logger=config["logger"],
+        max_epochs=config["max_epochs"],
+        num_sanity_val_steps=config["num_sanity_val_steps"],
+        strategy=config["strategy"],
+    )
+    # Train the model.
+    trainer.fit(
+        model, train_loader.train_dataloader(), val_loader.val_dataloader()
+    )
+    # Clean up temporary files.
+    tmp_dir.cleanup()
+
+
+def _get_peak_filenames(peak_dir: str) -> List[str]:
+    """
+    Get the peak file names in the given directory.
+
+    Parameters
+    ----------
+    peak_dir : str
+        The directory in which to find peak files.
+
+    Returns
+    -------
+    List[str]
+        The peak file names from the given directory.
+    """
+    return [
+        os.path.join(peak_dir, f)
+        for f in os.listdir(peak_dir)
+        if f.lower().endswith(".mgf")
+    ]
```

Comment:
**question:** Why not use `pathlib.Path.glob()` or `glob.glob()` here?

Reply:
Yes, makes sense. #62 does shell parameter extension to process multiple files rather than specifying input _directories_ (which precludes that training and testing are in the same directory, which was quite annoying imo), so that PR kinda does it how you suggest.